index.bs

```diff
@@ -583,36 +583,33 @@ interface ML {
 </script>
 
 The {{ML/createContext()}} method steps are:
-1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, then throw a "{{SecurityError!!exception}}" {{DOMException}} and abort these steps.
 1. Let |promise| be [=a new promise=].
+1. Return |promise| and run the following steps [=in parallel=].
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, then reject |promise| with a "{{SecurityError!!exception}}" {{DOMException}} and abort these steps.
 1. Let |context| be a new {{MLContext}} object.
-1. Switch on the method's first argument:
+1. Let |options| be the first argument.
+1. Switch on |option|:
     <dl class=switch>
-    <dt>{{MLContextOptions}}
+    <dt>{{MLContextOptions}} or {{undefined}}
     <dd>Set |context|.{{[[contextType]]}} to [=default-context|default=].
-    <dd>Set |context|.{{[[deviceType]]}} to the value of {{MLContextOptions}}'s {{deviceType}}.
-    <dd>Set |context|.{{[[powerPreference]]}} to the value of {{MLContextOptions}}'s {{powerPreference}}.
+    <dd>Set |context|.{{[[deviceType]]}} to the value of |option|'s {{deviceType}} or "[=device-type-cpu|cpu=]".
+    <dd>Set |context|.{{[[powerPreference]]}} to the value of |options|'s {{powerPreference}} or "[=power-preference-default|default=]".
 
     <dt>{{GPUDevice}}
     <dd>Set |context|.{{[[contextType]]}} to [=webgpu-context|webgpu=].
     <dd>Set |context|.{{[[deviceType]]}} to "[=device-type-gpu|gpu=]".
     <dd>Set |context|.{{[[powerPreference]]}} to "[=power-preference-default|default=]".
     </dl>
-1. Issue the following steps to a separate timeline:
-    1. If the User Agent can support the |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}, then:
-        1. Set |context|.{{MLContext/[[implementation]]}} to an implementation supporting |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}.
-        1. [=Resolve=] |promise| with |context|.
-    1. Else:
-        1. [=Resolve=] |promise| with a new {{NotSupportedError}}.
-1. Return |promise|.
+1. If the UA cannot support |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}, [=reject=] |promise| with a new {{NotSupportedError}} and abort these steps.
```

Comment:
The spec uses "user agent" in its lower case form elsewhere except in the context of `[[implementation]]` internal slot. I'd suggest using "user agent" here too and update the uppercase form to match.

Reply:
Yes, that's a bug, nowadays it's simply "user agent".